securedrop/models.py

```diff
@@ -212,7 +211,23 @@ def __init__(self, source: Source, filename: str) -> None:
     def __repr__(self) -> str:
         return '<Submission %r>' % (self.filename)
 
-    def to_json(self) -> 'Dict[str, Union[str, int, bool]]':
+    @property
+    def is_file(self) -> bool:
+        return self.filename.endswith("doc.gz.gpg") or self.filename.endswith("doc.zip.gpg")
+
+    @property
+    def is_message(self) -> bool:
+        return self.filename.endswith("msg.gpg")
+
+    def to_json(self) -> "Dict[str, Union[str, int, bool]]":
+        seen_by = {
+            f.journalist.uuid for f in SeenFile.query.filter(SeenFile.file_id == self.id)
+            if f.journalist
```

Comment:
i see, so this won't include the fact that there is a deleted journalist who saw a file. gonna keep reviewing and will cycle back to this. could be an issue because we still send "deleted" uuids for replies when journalists are deleted. this takes a different course.

Reply:
It does not. I didn't see any way for sensible inclusion of those records in `seen_by` here. We do have `Submission.seen`, which will reflect those, and until we start keeping deleted journalists around, we can't distinguish between the records for the submission in the seen tables anyway, so since the only purpose of those rows at this point is establishing the global seen state, I thought `Submission.seen` the right answer.

Reply:
There are also replies we have to worry about which don't have a global seen state, but I definitely see why you made this choice. Agreed that implementing https://github.com/freedomofpress/securedrop/issues/5467 + global user account for data migration will make this work fine.

Reply:
Replies do have a global seen state: if one exists, its creator has seen it. `SeenReply` can tell us who else has seen it, but only if  `journalist_id` is not null, so again, I didn't see any point in including records where it is.